dl1_data_handler/image_mapper.py

```diff
@@ -1,744 +1,1112 @@
-            "HESS-I": (72, 72, 1),
-            "HESS-II": (104, 104, 1),
-        }
+        """
+        Parameters
+        ----------
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent : ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        """
 
         # Camera types
-        if camera_types:
-            self.camera_types = []
-            for camera_type in camera_types:
-                if camera_type in self.image_shapes:
-                    self.camera_types.append(camera_type)
-                else:
-                    logger.error("Camera type {} isn't supported.".format(camera_type))
-        else:
-            self.camera_types = [cam for cam in self.image_shapes]
-
-        # Mapping method
-        if mapping_method is None:
-            mapping_method = {}
-        self.mapping_method = {
-            **{c: "oversampling" for c in self.camera_types},
-            **mapping_method,
-        }
-
-        # Interpolation image shape
-        if interpolation_image_shape is None:
-            interpolation_image_shape = {}
-        self.interpolation_image_shape = {
-            **{c: self.image_shapes[c] for c in self.camera_types},
-            **interpolation_image_shape,
-        }
-
-        # Padding
-        if padding is None:
-            padding = {}
-        self.padding = {**{c: 0 for c in self.camera_types}, **padding}
-
-        # Mask interpolation
-        self.mask = True if mask_interpolation else False
-
-        # Pixel positions, number of pixels, mapping tables and index matrixes initialization
-        self.pixel_positions = {}
-        self.num_pixels = {}
-        self.mapping_tables = {}
-        self.index_matrixes = {}
-
-        for camtype in self.camera_types:
-            # Get a corresponding pixel positions
-            if pixel_positions is None:
-                try:
-                    from ctapipe.instrument.camera import CameraGeometry
-                except ImportError:
-                    raise ImportError(
-                        "The `ctapipe.instrument.camera` python module is required, if pixel_positions is `None`."
+        self.geometry = geometry
+        self.camera_type = self.geometry.name
+        self.n_pixels = self.geometry.n_pixels
+        # Rotate the pixel positions by the pixel to align
+        self.geometry.rotate(self.geometry.pix_rotation)
+
+        self.pix_x = np.around(
+            self.geometry.pix_x.value, decimals=constants.decimal_precision
+        )
+        self.pix_y = np.around(
+            self.geometry.pix_y.value, decimals=constants.decimal_precision
+        )
+
+        self.x_ticks = np.unique(self.pix_x).tolist()
```

Comment:
Am I right you assume regularly spaced pixels for any kind of camera geometry? If not, did you test any geometry with shuffle step (e.g. square pixels where rows are shifted by e.g. 25%?)